### ISSUE TITLE
`Pin versions in pip` linting error

### ISSUE THREAD

**Issue by jamesstonehill** [hadolint/hadolint#366]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

When my Dockerfile looks like this:
```
RUN pipenv run python setup.py install
```
I get this:
```
$ hadolint Dockerfile
Dockerfile:1 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

I shouldn't get a linter error in this case as there is no `pip install` command in that line.

### Steps to reproduce the behavior

```bash
echo "RUN pipenv run python setup.py install" > Dockerfile
hadolint Dockerfile
```

**Comment by ferrarimarco** [comment]
I think this is due to https://github.com/hadolint/hadolint/blob/73081a9f21799906891c2133b8119cd0c456ded3/src/Hadolint/Rules.hs#L1125

We're having the same issue in our CI builds:

```shell
[Dockerfile:121 DL3042 Avoid use of cache directory with pip. Use `pip install --no-cache-dir <package>`]
```

triggered by

```dockerfile
RUN pipenv install --system
```

**Comment by rodrigoaguilera** [comment]
I am also affected by this. I believe the reason it checks the `pip` prefix is because some systems have `pip` and/or `pip3` (maybe pip2) but as far as I know those are the only cases so the code should look for a defined list of `pip` variants instead of any command starting with `pip`.

Sadly I don't know enough haskell to propose a PR
